### ISSUE TITLE
Dependency conflicts

### ISSUE THREAD

**Issue by Monokai** [langchain-ai/langchainjs#3224]
I'm running langchain in a project that already contains a mongodb. When I install langchain in this project, I'm getting depency conflicts:

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: langchain@0.0.185
npm ERR! Found: mongodb@6.2.0
npm ERR! node_modules/mongodb
npm ERR!   mongodb@"^6.2.0" from the root project
npm ERR!   peer mongodb@">= 5.1.0 < 7" from connect-mongo@5.1.0
npm ERR!   node_modules/connect-mongo
npm ERR!     connect-mongo@"^5.1.0" from the root project
npm ERR!   1 more (mongoose)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peerOptional mongodb@"^5.2.0" from langchain@0.0.185
npm ERR! node_modules/langchain
npm ERR!   langchain@"^0.0.185" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: mongodb@5.9.1
npm ERR! node_modules/mongodb
npm ERR!   peerOptional mongodb@"^5.2.0" from langchain@0.0.185
npm ERR!   node_modules/langchain
npm ERR!     langchain@"^0.0.185" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

Looking at this, it seems it's an optional dependency. As far as I know, I don't use the mongodb functionalities of lanchain itself.

As I understand, I can omit the optional dependencies by running `npm install --omit=optional`. This unfortunately gives the same error. I'm aware of `npm install --legacy-peer-deps`, but I'd rather not do this as the rest of my package dependencies work fine and I don't want to break that.

Ideally langchain would be a bit less strict here. Or maybe I'm missing something. Any ideas?

**Comment by ElectricCodeGuy** [comment]
Ran into the same issue since version 0.0.183 but with @vercel/kv:

npm ERR!   @vercel/kv@"^1.0.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peerOptional @vercel/kv@"^0.2.3" from langchain@0.0.185

**Comment by jacoblee93** [comment]
This behavior for unused peer deps is a bit silly by `npm` and can be safely ignored with your command above. `yarn` and `pnpm` both don't have this problem.

We can set `*` or `>=` as a dep but that has caused problems for us in the past because it can result in peer deps updating and creating version conflicts with other peer deps. It still might be the best solution for now if this is becoming more common...

**Comment by Monokai** [comment]
@jacoblee93 I understand your point of view, I also struggled with peer deps before. But as it stands langchain is the only package that prevents me from doing a normal `npm install`. I can't be the only one that want to include langchain in an existing project.

But whenever your project includes a newer package than the *optional* dependency that langchain defines in its peer deps, regardless of whether you use this optional dependency with langchain, this prevents you from doing a normal `npm install`.

Maybe make the optional peer dependencies into proper, separate optional plugin packages?

**Comment by jacoblee93** [comment]
Yes, long term something like that is going to be the best solution, but a significant amount of work. Maybe using >= is a good compromise in the short run

**Comment by kabdelkareem** [comment]
same issue here.

**Comment by jacoblee93** [comment]
We have actively started doing the above - we will be factoring more and more integrations into their own packages/community and the current plan is to fully remove the community dep in the next minor release.

**Comment by ElectricCodeGuy** [comment]
> We have actively started doing the above - we will be factoring more and more integrations into their own packages/community and the current plan is to fully remove the community dep in the next minor release.

Thanks for the work! it works perfectly fine with the new Pinecone from the update you just pushed! :)

**Comment by willydavid1** [comment]
With npm, I get the same error. 'npm i langchain'

With 'yarn add langchain,' everything installs correctly.

**Comment by jacoblee93** [comment]
It's because of `npm`'s stricter peer dependency validation :( we suggest using `yarn` or `pnpm` but will be working to address this more in a future 0.2.0 (which will be breaking).

**Comment by mdnazim5** [comment]
@jacoblee93 : I see a similar problem with the latest version of langchain 0.1.14

![image](https://github.com/langchain-ai/langchainjs/assets/65829249/32882b35-8d6c-4cbb-a006-23db6861ef13)

**Comment by jacoblee93** [comment]
For now, you can go down to 0.1.13 @mdnazim5.

**Comment by jacoblee93** [comment]
@mdnazim5 langchain@0.1.15 should remove this dep.

**Comment by Monokai** [comment]
This was closed by the bot as not planned, but I'm opening it up again, quoting @jacoblee93:

> We have actively started doing the above - we will be factoring more and more integrations into their own packages/community and the current plan is to fully remove the community dep in the next minor release.
